Fixes #600 Support Experiment Item Type

richbenmintz opened the pull request:
## Description

Add MLExperiment type to supported item types

## Linked Issue https://github.com/microsoft/fabric-cicd/issues/600

<!-- 
REQUIRED: This PR must be linked to an issue via the PR title format.
PR Title MUST follow this exact format: "Fixes #123 - Short Description"

Use the appropriate action word:
- Fixes #123 (for bug fixes)
- Closes #456 (for features)  
- Resolves #789 (for other changes)

Example: "Fixes #520 - Add Python version requirements to documentation"
-->
Fixes #600 - Support Experiment Item Type

shirasassoon commented on devtools/debug_local.py:
```diff
@@ -48,6 +48,7 @@
     "KQLQueryset",
     "Reflex",
     "Eventstream",
+    "MLExperiment",
```
Revert all changes to original in this file, it is not necessary to include the MLExperiment item type in scope in the debug file.

shirasassoon reviewed (changes requested):
@richbenmintz Added some comments, overall, the code looks good but is missing some key parts (see comments on publish.py). Please confirm you tested these changes and it works as expected. Once the comments are addressed, we can review again.

richbenmintz commented on src/fabric_cicd/publish.py:
added

shirasassoon replied:
For consistency, please add as the first item in the list (following reverse publish order convention).

shirasassoon commented on docs/how_to/item_types.md:
```diff
@@ -114,6 +114,10 @@
 -   **Unpublish** - a warning is shown for any default Semantic Models created by the Mirror Database. This is a current limitation of the Fabric API and can be ignored.
 
+## ML Experiments
+
```
Please add a point for parameterization.

richbenmintz replied:
done

shirasassoon commented:
Please update the branch.

shirasassoon commented on devtools/debug_local.py:
```diff
@@ -27,7 +27,7 @@
 
 # The defined environment values should match the names found in the parameter.yml file
-workspace_id = "8f5c0cec-a8ea-48cd-9da4-871dc2642f4c"
+workspace_id = "f0e3fa10-e2d7-4ce6-a508-4e6e16ba3a27"
```
Revert this change to the original GUID

shirasassoon commented:
> The experiment deployer will not create the notebook correct?

When you create a new experiment there is an option to "start with a template" and that will automatically create a new notebook. Once the notebook is create it, run it and once complete, return to the experiment and refresh it. Commit both to source control and you will see the notebook dependency present in the JSON.

shirasassoon commented:
@richbenmintz Need any further assistance?

richbenmintz commented:
@shirasassoon, I updated my experiment and added the template items, however when commited to devops they are not included. I also tested by creating a new experiment and adding a template, metadata.json blank

<img width="589" height="221" alt="image" src="https://github.com/user-attachments/assets/6bf06402-5df3-4993-9695-3f53d446a55f" />

shirasassoon commented:
> @shirasassoon, I updated my experiment and added the template items, however when commited to devops they are not included. I also tested by creating a new experiment and adding a template, metadata.json blank
> 
> <img alt="image" width="589" height="221" src="https://private-user-images.githubusercontent.com/12468093/519223402-6bf06402-5df3-4993-9695-3f53d446a55f.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjQ1MDQ3NzEsIm5iZiI6MTc2NDUwNDQ3MSwicGF0aCI6Ii8xMjQ2ODA5My81MTkyMjM0MDItNmJmMDY0MDItNWRmMy00OTkzLTk2OTUtM2Y1M2Q0NDZhNTVmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTExMzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMTMwVDEyMDc1MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQ3YWVhZjYyMjA1NjgyM2YzMWU3MGY3ZjYwMzQ0YmFiY2JlNThiYWIwY2M0YzAxYjViM2Y3NDBjMWNiYjAyZTImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.oDFu7Gj5E4sW6Z-6WWx9xikJafjHnRtqZNEDOYbis1s">

We can leave it as is then
